### PR TITLE
Staging the GCE Cloud Provider

### PR DESCRIPTION
**Pull request by cheftako**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Staging the GCE Cloud Provider as part of KEP [20190125-removing-in-tree-providers](https://github.com/kubernetes/enhancements/blob/master/keps/sig-cloud-provider/20190125-removing-in-tree-providers.md). Staging repo setup here https://github.com/kubernetes/legacy-cloud-providers
Moves the GCE cloud provider implementation to staging.
This is in preparation for moving the cloud provider code out of tree entirely.
However we need it in staging while the code needs to be consumed both in/out of tree.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```
NONE
```

**Release note**:
```release-note
None
```

### CODE REVIEW

**Review by BenTheElder — APPROVED** [review]
/lgtm
/approve

File: staging/publishing/import-restrictions.yaml [hunk 1]
```diff
@@ -207,6 +207,7 @@
   allowedImports:
   - k8s.io/api
   - k8s.io/apimachinery
+  - k8s.io/apiserver
```
Comment (liggitt) [inline staging/publishing/import-restrictions.yaml]: curious what required this one

Comment (liggitt) [reply staging/publishing/import-restrictions.yaml]: if this link is easy to snip, I'd recommend doing so. apiserver pulls in a lot of transitive dependencies

Comment (cheftako) [reply staging/publishing/import-restrictions.yaml]: All forms of https://github.com/kubernetes/kubernetes/blob/master/pkg/cloudprovider/providers/gce/gce_disks.go#L149. Essentially checking if certain cloud features for disk are enabled. 
Will look at snipping it tomorrow.

Comment (liggitt) [reply staging/publishing/import-restrictions.yaml]: https://github.com/kubernetes/kubernetes/pull/74021 relocates the feature gate package out of k8s.io/apiserver for just this purpose.

Comment (liggitt) [reply staging/publishing/import-restrictions.yaml]: component-base is a much more reasonable dep

Comment (cheftako) [reply staging/publishing/import-restrictions.yaml]: > component-base is a much more reasonable dep

Agreed. @andrewsykim Do you have time to push #74021 forward? Would you like help?

Comment (andrewsykim) [reply staging/publishing/import-restrictions.yaml]: Sorry, that one slipped my radar. I can get to it today :)

Comment (liggitt) [reply staging/publishing/import-restrictions.yaml]: actually, looks like GCERegionalPersistentDisk is the only feature checked, and that went GA in 1.13.

can we lock that feature on and remove all uses of it in gce? if so, I think we can drop the utilfeature check entirely and skip waiting on https://github.com/kubernetes/kubernetes/pull/74021

Comment (liggitt) [reply staging/publishing/import-restrictions.yaml]: removed use of the feature gate in https://github.com/kubernetes/kubernetes/pull/77412

Comment (liggitt) [reply staging/publishing/import-restrictions.yaml]: Merged, this should be in the clear once it rebases

**Comment by liggitt** [comment]
looks like it needs a rebase and `hack/update-vendor.sh` to drop the apiserver dep from the go.mod, then lgtm

**Comment by BenTheElder** [comment]
🎉

*From: *Kubernetes Prow Robot <notifications@github.com>
*Date: *Tue, May 7, 2019 at 5:54 PM
*To: *kubernetes/kubernetes
*Cc: *Benjamin Elder, Assign

Merged #77374 <https://github.com/kubernetes/kubernetes/pull/77374> into

**Pull request merged**